lib/Dialect/RTL/Ops.cpp

```diff
@@ -427,7 +427,33 @@ void AndOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
         return success();
       }
 
-      /// TODO: and(x, and(...)) -> and(x, ...) -- flatten
+      // and(x, and(...)) -> and(x, ...) -- flatten
+      for (size_t i = 0; i != size; ++i) {
+        if (!inputs[i].hasOneUse())
+          continue;
```

Comment:
Nice use of continue to keep the flow consistent.  You might do the same thing below to reduce nesting:

```
auto andOp = ...
if (!andOp) continue;
```